app/controllers/story.controller.js

```diff
@@ -200,3 +200,72 @@ exports.findAllByUser = async (req, res) => {
     }
 }
 
+
+// Add a chat to a story
+
+exports.addChat = async (req, res) => {
+
+    if (req.body.message === undefined) {
+        const error = new Error("Message cannot be empty for chat!");
+        error.statusCode = 400;
+        res.status(400).send({
+            message:
+                error.message || "Some error occurred while adding chat to story.",
+        });
+
+    }
+    const id = req.params.id;
+
+    try {
+        const chat = {
+            role: "User",
+            message: req.body.message,
+            storyId: id,
+        };
+        const history = await Chat.findAll({ where: { storyId: id } });
+
+
+        const chohereResponse = await axios.post('https://api.cohere.com/v1/chat', {
+            model: "command-r-plus",
+            message: req.body.message,
+            temperature: 0.3,
+            chat_history: history,
+            stream: false,
+            connectors: [{"id":"web-search"}]
+        }, {
+            headers: {
+                'Authorization': `Bearer ${process.env.COHERE_API_KEY}`,
+                'Content-Type': 'application/json'
+            }
+        });
+
+        const userMessage = await Chat.create(chat);
+        const chatMessage = await Chat.create({ role: "Chatbot", message: chohereResponse.data.text, storyId: id });
+        history.push(userMessage)
+        history.push(chatMessage)
+        res.send(history);
+    } catch (err) {
+        console.log(err);
+        res.status(500).send({
+            message: "Error adding chat to Story with id=" + id,
```

Comment:
include details of the `err` value in the response message?

Reply:
Added err is response.